FreeIT/src/lesson_10/Application.java

```diff
@@ -121,66 +119,57 @@ public int compare(Object o1, Object o2) {
 		System.out.println(a);
 	}
 
-	private void sort2() {
+	private void sort3() {
 		System.out.println(libr);
 	}
 
-	private void sort3() {
-		ListIterator<Book> iter1 = libr.getBooks().listIterator();
+	private void sort2() {
 		ArrayList<Book> b1 = new ArrayList<>();
-		while (iter1.hasNext()) {
-			iter1.next();
-		}
+		b1.add(libr.getBooks().get(libr.getBooks().size() - 1));
+		ListIterator<Book> iter1 = libr.getBooks().listIterator(libr.getBooks().size() - 1);
 		while (iter1.hasPrevious()) {
 			b1.add(iter1.previous());
 		}
 		System.out.println(b1);
 	}
 
 	private void addingBook() {
-		System.out.println("Сейчас будем вводить данные для создания новой книги. Введите id (целое число):");
-		Scanner sc = new Scanner(System.in);
-		int id = 1;
-		try {
-			id = sc.nextInt();
-		} catch (InputMismatchException e) {
-			System.out.println("Вы должны ввести число! Попробуйте еще раз!");
-			addingBook();
-			return;
-		}
+		System.out.println("Сейчас будем вводить данные для создания новой книги.");
+		Integer id = 0;
+		do {
+			System.out.println("Введите id (целое положительное число):");
+			id = inputFromConcole();
```

Comment:
create please logic for Book ID auto generation and remove direct assigning of book ID